src/modules/musig/tests_impl.h

```diff
@@ -1045,7 +1045,7 @@ void musig_test_vectors_noncegen(void) {
 
         CHECK(secp256k1_ec_pubkey_parse(ctx, &pk, c->pk, sizeof(c->pk)));
         CHECK(secp256k1_musig_nonce_gen(ctx, &secnonce, &pubnonce, c->rand_, sk, &pk, msg, keyagg_cache_ptr, extra_in) == 1);
-        CHECK(secp256k1_memcmp_var(&secnonce.data[4], c->expected, sizeof(secnonce)-4) == 0);
+        CHECK(secp256k1_memcmp_var(&secnonce.data[4], c->expected, 2*32) == 0);
```

Comment:
I think this should be kept `sizeof(secnonce) - 4`. We want to compare the entire secnonce, which is 97 bytes now in the BIP.

Reply:
The secnonce in this implementation is not 97 bytes (minus magic) because it contains a (decompressed) pubkey. I added a line that tests that the public key has been appended to the secnonce.